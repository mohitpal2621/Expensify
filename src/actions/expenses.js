import { v4 as uuid } from "uuid";

// ADD_EXPENSE action generator
export const addExpense = ({ description='', note='', amount=0, createdAt=0 }) => ({
    type: 'ADDEXPENSE',
    expense: {
        id: uuid(),
        description,
        note,
        amount,
        createdAt
    }
});

// REMOVE_EXPENSE action generator
export const removeExpense = (id) => ({
    type: 'REMOVEEXPENSE',
    id
});

// EDIT_EXPENSE action generator
export const editExpense = (id, updates) => ({
        type: 'EDITEXPENSE',
        id,
        updates
});