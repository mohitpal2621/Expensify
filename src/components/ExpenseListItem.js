import React from "react";
import moment from "moment";
import curr from "../format/currFormat";
import { Link } from "react-router-dom";


const ExpenseListItem = ({ id, description, amount, createdAt }) => (
    <Link className="list-item" to={`/edit/${id}`}>
        <div className="list-item__div">
            <h3 className="list-item__title">{description}</h3>
            <span className="list-item__sub-title">{moment(createdAt).format('MMMM Do, YYYY')}</span>
        </div>
        <h3 className="list-item__data">{curr.format(amount).replace("INR", "Rs.")}</h3>
    </Link>
)

export default ExpenseListItem;