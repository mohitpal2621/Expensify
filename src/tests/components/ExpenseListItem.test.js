import React from "react";
import { render, screen } from '@testing-library/react';
import ExpenseListItem from "../../components/ExpenseListItem";
import expenses from '../fixtures/expenses';
import { MemoryRouter } from 'react-router-dom';

test("Should render a list item with given expense", () => {
    const {container} = render(<MemoryRouter><ExpenseListItem {...expenses[0]} /></MemoryRouter>);
    expect(container).toMatchSnapshot();
})