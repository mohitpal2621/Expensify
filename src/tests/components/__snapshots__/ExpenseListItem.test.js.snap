// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render a list item with given expense 1`] = `
<div>
  <a
    class="list-item"
    href="/edit/1"
  >
    <div
      class="list-item__div"
    >
      <h3
        class="list-item__title"
      >
        Gum
      </h3>
      <span
        class="list-item__sub-title"
      >
        January 1st, 1970
      </span>
    </div>
    <h3
      class="list-item__data"
    >
      Rs.Â 140.00
    </h3>
  </a>
</div>
`;
